src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,145 @@
 package infrastructure.content;
 
+import org.json.JSONObject;
+import networking.CommunicatorFactory;
+import networking.ICommunicator;
+import java.util.HashMap;
+
+/**
+ * This class ContentCommunicator implements the IContentCommunicator interface and all the 
+ * methods inside it. The ideal purpose of this class is to initialise any new client, send 
+ * message from UI to networking module, handle client exit with respect to content module
+ * and handle the subscribers of content module.
+ * 
+ * @author Badal Kumar (111701008)
+ */
 public class ContentCommunicator implements IContentCommunicator{
-    
-	@Override
+	/**
+	 * This variable will store the username/userid of client as a String.
+	 */
+	private String userName;
+	
+	/**
+	 * This variable will store the image of client in String format.
+	 */
+	private String userImage;
+	
+	/**
+	 * This variable will store the server's ipAddress as a String.
+	 */
+	private String serverIPAddress;
+	
+	/**
+	 * This variable will store the port number of the board which the client is accessing.
+	 */
+	private int port = 0;
+	
+	/**
+	 * Creating a HashMap to store userName and userImage of all active clients of a particular board.
+	 * Both userName and userImage are of String type
+	 */
+	private HashMap<String, String> mapImage = new HashMap<String, String>();
+	
+	/**
+	 * Creating a HashMap to store the identifier and handler of subscriber of this module,
+	 * identifier is of type String while handler is of type IContentNotificationHandler.
+	 */
+	private HashMap<String, IContentNotificationHandler> mapHandler = new HashMap<String, IContentNotificationHandler>();
+	
+	/**
+	 * This variable will store the communicator of networking module to send data over the network.
+	 */
+	private ICommunicator communicator;
+	

```

Comment:
There needs to be a constructor for ContentCommunicator() which subscribes to networking notification

Reply:
Either you or Talha needs to do it. On that note, currently Talha's class is not initialised anywhere and so can't be used. Please speak with him on whether it can be initialised here or if there is some way better.